src/main/java/hudson/plugins/throttleconcurrents/ThrottleJobProperty.java

```diff
@@ -491,9 +496,73 @@ public void load() {
         private synchronized void initThrottledPipelines() {
             if (throttledPipelinesByCategory == null) {
                 throttledPipelinesByCategory = new TreeMap<>();
+            } else if (throttledPipelinesByCategory.entrySet().stream()
+                    .anyMatch(e -> !(e.getValue() instanceof CopyOnWriteMap.Tree))) {
+                // if any of the nested maps are not copy-on-write tree maps, convert the whole data
+                // structure.
+                LOGGER.log(
+                        Level.INFO,
+                        "Migrating throttled pipelines by category to copy-on-write data structures.");
+                LOGGER.log(Level.FINE, "Original values: {0}", throttledPipelinesByCategory);
+                // For consistency, the type of map returned below should match that of the
+                // initThrottledPipelines method.
+                throttledPipelinesByCategory =
+                        throttledPipelinesByCategory.entrySet().stream()
+                                .collect(
+                                        Collectors.toMap(
+                                                Map.Entry::getKey,
+                                                e ->
+                                                        convertToCopyOnWriteDataStructures(
+                                                                e.getValue()),
```

Comment:
This formatting feels a bit odd...

Maybe we rework `convertToCopyOnWriteDataStructures` to be `convertValueToCopyOnWriteDataStructures` and make it receive a `Map.Entry` where it unwraps the entry value, because then we could call it here like this:

`this::convertValueToCopyOnWriteDataStructures`

This is a bit nit-picky, but let me know what you think.

Reply:
I'm going to open this suggestion in a subsequent PR into this branch, so I can show you what I mean before we go with it.